src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmExecutor.java

```diff
@@ -0,0 +1,105 @@
+import java.util.zip.GZIPInputStream;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.alliancegenome.curation_api.dao.AffectedGenomicModelDAO;
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException;
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException.ObjectUpdateExceptionData;
+import org.alliancegenome.curation_api.model.entities.AffectedGenomicModel;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFile;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFileHistory;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkManualLoad;
+import org.alliancegenome.curation_api.model.ingest.dto.AffectedGenomicModelDTO;
+import org.alliancegenome.curation_api.model.ingest.dto.IngestDTO;
+import org.alliancegenome.curation_api.response.APIResponse;
+import org.alliancegenome.curation_api.response.LoadHistoryResponce;
+import org.alliancegenome.curation_api.services.AffectedGenomicModelService;
+import org.alliancegenome.curation_api.util.ProcessDisplayHelper;
+
+import lombok.extern.jbosslog.JBossLog;
+
+@JBossLog
+@ApplicationScoped
+public class AgmExecutor extends LoadFileExecutor {
+
+    @Inject AffectedGenomicModelDAO affectedGenomicModelDAO;
+
+    @Inject AffectedGenomicModelService affectedGenomicModelService;
+
+    public void runLoad(BulkLoadFile bulkLoadFile) {
+
+        try {
+            BulkManualLoad manual = (BulkManualLoad) bulkLoadFile.getBulkLoad();
+            log.info("Running with: " + manual.getDataType().name() + " " + manual.getDataType().getTaxonId());
+            
+            IngestDTO ingestDto = mapper.readValue(new GZIPInputStream(new FileInputStream(bulkLoadFile.getLocalFilePath())), IngestDTO.class);
+            List<AffectedGenomicModelDTO> agms = ingestDto.getAgmIngestSet();
+            String taxonId = manual.getDataType().getTaxonId();
+            
+            if (agms != null) {
+                bulkLoadFile.setRecordCount(agms.size() + bulkLoadFile.getRecordCount());
+                bulkLoadFileDAO.merge(bulkLoadFile);
+                
+                trackHistory(runLoad(taxonId, agms), bulkLoadFile);
+            }
+            
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    
+    // Gets called from the API directly
+    public APIResponse runLoad (List <AffectedGenomicModelDTO> agms) {
+        List<String> taxonIds = agms.stream()
+                .map( agmDTO -> agmDTO.getTaxon() ).distinct().collect( Collectors.toList() );
+        return runLoad(taxonIds, agms);
+    }
+        
+    public APIResponse runLoad(String taxonId, List<AffectedGenomicModelDTO> agms) {
+        List<String> taxonIds = new ArrayList<String>();
+        taxonIds.add(taxonId);
+        return runLoad(taxonIds, agms);
+    }
+            
+    public APIResponse runLoad(List<String> taxonIds, List<AffectedGenomicModelDTO> agms) {
```

Comment:
What case does this cover? Just wondering why we would have a list of taxons?

Reply:
SGD has multiple taxon IDs in their AGM submission